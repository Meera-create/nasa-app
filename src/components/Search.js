import React,{ useState }  from 'react';
import '../styles/search.css'

const Search=()=>{
const[value,setValue]=useState();


    return (
        <>
        <form className="form">
        <input className="searchBox" type="text" onChange={(e)=> setValue(e.target.value)}/>

        <button className="button" type="submit">SEARCH</button>
        </form>
        </>
    );
};


export default Search;


//onchange will save useres search query every time there is a change in input box