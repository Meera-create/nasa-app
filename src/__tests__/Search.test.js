import { render, screen } from '@testing-library/react';
import Search from '../components/Search';



describe("tests search box",()=>{
    const{asFragment}=render(<Search />);

    it("renders correctly",()=>{
        expect(asFragment()).toMatchSnapshot();
    });
    
});