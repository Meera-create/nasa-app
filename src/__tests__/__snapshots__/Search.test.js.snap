// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests search box renders correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <input
      class="searchBox"
      type="text"
    />
    <button
      class="button"
      type="submit"
    >
      SEARCH
    </button>
  </form>
</DocumentFragment>
`;
